From 4a8245ec2e9d48e464488477a3e7ed234009c216 Mon Sep 17 00:00:00 2001
From: Peter Oh <peter.oh@bowerswilkins.com>
Date: Tue, 29 May 2018 14:39:16 -0700
Subject: [PATCH 12/18] mesh: allow mesh to send channel switch request

add mesh type to nl80211 channel switch request,
so mesh is able to send the request to kernel drivers.

Signed-off-by: Peter Oh <peter.oh@bowerswilkins.com>
---
 src/drivers/driver_nl80211.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/src/drivers/driver_nl80211.c
+++ b/src/drivers/driver_nl80211.c
@@ -8685,7 +8685,8 @@ static int nl80211_switch_channel(void *
 	}
 
 	if ((drv->nlmode != NL80211_IFTYPE_AP) &&
-	    (drv->nlmode != NL80211_IFTYPE_P2P_GO))
+	    (drv->nlmode != NL80211_IFTYPE_P2P_GO) &&
+	    (drv->nlmode != NL80211_IFTYPE_MESH_POINT) )
 		return -EOPNOTSUPP;
 
 	/*
